### ISSUE TITLE
Allow ability to only have one row selected at a time

### ISSUE THREAD

**Issue by homesar** [hjalmers/angular-generic-table#72]
I thought this would be included as part of #53 but I guess the 'single/multiple' referred to the ability to select one/multiple rows and add them to the list of selected rows.  I would like the ability for the grid to only ever allow one row to be selected at a time.

**Comment by hjalmers** [comment]
Yeah good point and I agree, same goes for expanding rows. Will make sure it's part of the next release:)

**Comment by abonsignore** [comment]
Hey Robert,
I also have a need for single selection. Proves useful in Master Detail relationships.
Great to get this implemented.

In the interim... I did the following:

```
export interface GtOptions {
  cache?: boolean;
  debounceTime?: number;
  csvDelimiter?: string;
  stack?: boolean;
  lazyLoad?: boolean;
  highlightSearch?: boolean;
  rowSelection?: boolean;
  rowSelectionSingle?: boolean;  //**** Added here *****
}

export class GenericTableComponent ...

public toggleSelect(row:GtRow) {
    //*****Added the next 3 lines ****
    if (this.gtOptions.rowSelectionSingle) {
      this.deselectAllRows();
    }
    this._toggleRowProperty(row,'isSelected');
  }

```
... and in Markup

```
<generic-table [gtClasses]="'table-hover'" #myTable [gtSettings]="configObject.settings" 
        [gtFields]="configObject.fields"
        [(gtData)]="configObject.data" [gtInfo]="configObject.info" [gtRowComponent]="expandedRow" 
      (gtEvent)="trigger($event)"
        [gtOptions]="{stack:true, highlightSearch:true, rowSelection:true, rowSelectionSingle:true}">
</generic-table>

```

**Comment by hjalmers** [comment]
@abonsignore, thanks for the input, I actually had something similar in mind:) Shouldn't take to long to add this, however we're in the middle of changing the module structure for the project to make it easier to develop extensions and plugins in the future. So next release will most likely introduce @angular-generic-table/core and @angular-generic-table/pagination etc. but as soon as I'm finished with it I'll add this, hopefully during this week (without making any promises). :)

**Comment by hjalmers** [comment]
This feature will be available in next version (v4.1.0). Disable multi row selection by setting `rowSelectionAllowMultiple` to `false` same goes for expanding rows, disable multiple expanded rows by setting `rowExpandAllowMultiple` to `false`
